torch/csrc/jit/codegen/cuda/partition.cpp

```diff
@@ -428,7 +440,7 @@ bool isFusibleCudaFusionGroup(const Node* fusion, const Node* node) {
   bool fused = false;
   // TODO: lift the restriction of not fusing producer containing reduction when
   //       we have proper scheduling.
-  if (isFusibleCudaFusionGroup(node)) {
+  if (isFusibleNode(node)) {
```

Comment:
Yes!

This one is a little bit funny here. I'm assuming that if we want to fuse `node` with `fusion`, we are going to get a consistent `cuda` device in the end.
The consideration here is:
1. We don't want to fuse a single node that takes cpu scalar inputs and produces a cpu scalar output.
2. We do want to fuse node with mixed input (cuda tensor + cpu scalar inputs)
3. In the case where we have an operation only on cpu scalar inputs that leads into another operation with mixed device, we still want to fuse them.

So this means, when we want to construct a single node fusion `isFusibleCudaFusionGroup(const Node* node)`, we do want to make sure that one of the inputs are `cuda`, since it doesn't make sense for us to fuse cpu operations.
Meanwhile, if we fuse `node` into an existing fusion `isFusibleCudaFusionGroup(const Node* fusion, const Node* node)`, we don't need to check device of `node` since it's merged to `fusion`, which will give us an output tensor on cuda.

Reply:
Having said that, you mentioned about horizontal fusion earlier, which I think would actually break it here. :stuck_out_tongue: 

I need to disable horizontal fusion on cpu_scalar_tensor, which I think I forgot. Let me double check that one and add a test for it.

FYI, for the three cases I mentioned earlier, you can look at our python test for reference.

Reply:
Marking _**TODO**_ for myself